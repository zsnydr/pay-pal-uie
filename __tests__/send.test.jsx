import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Send from '../public/send/send';

describe('Send (Snapshot)', () => {
  it('Send renders new transaction page', () => {
    const component = renderer.create(<Send />);
    const send = component.toJSON();
    expect(send).toMatchSnapshot();
  });
});

xdescribe('Send', () => {
  let send;
  beforeEach(() => {
    send = mount(<Send />);
  });

  it('Input change calls changeHandler', () => {
    const textInput = send.find('TextInput').first();
    const form = textInput.find('input').first();
    form.simulate('change', { target: { value: 'z@s.com' } });
    send.update();
    const value = textInput.props().value;
    expect(value).toEqual('z@s.com');
  });
});
