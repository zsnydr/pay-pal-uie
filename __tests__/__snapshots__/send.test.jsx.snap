// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send (Snapshot) Send renders new transaction page 1`] = `
<div
  className="send"
>
  <div
    className="header"
  >
    Send Money
  </div>
  <div
    className="send-container"
  >
    <div
      className="send-forms"
    >
      <div
        className="text-input"
      >
        <label
          htmlFor="recipient"
        >
          To: 
          <input
            id="recipient"
            name="recipient"
            onChange={[Function]}
            type="text"
          />
        </label>
      </div>
      <div
        className="text-input"
      >
        <label
          htmlFor="amount"
        >
          Amount: $
          <input
            id="amount"
            name="amount"
            onChange={[Function]}
            type="text"
          />
        </label>
        <select
          onChange={[Function]}
        >
          <option
            value="USD"
          >
            USD
          </option>
          <option
            value="EUR"
          >
            EUR
          </option>
          <option
            value="CRC"
          >
            CRC
          </option>
          <option
            value="GBP"
          >
            GBP
          </option>
          <option
            value="ILS"
          >
            ILS
          </option>
          <option
            value="INR"
          >
            INR
          </option>
          <option
            value="JPY"
          >
            JPY
          </option>
          <option
            value="KRW"
          >
            KRW
          </option>
          <option
            value="NGN"
          >
            NGN
          </option>
          <option
            value="PHP"
          >
            PHP
          </option>
          <option
            value="PLN"
          >
            PLN
          </option>
          <option
            value="PYG"
          >
            PYG
          </option>
          <option
            value="THB"
          >
            THB
          </option>
          <option
            value="UAH"
          >
            UAH
          </option>
          <option
            value="VND"
          >
            VND
          </option>
        </select>
      </div>
      <div
        className="text-input"
      >
        <label
          htmlFor="message"
        >
          Message (optional): 
          <br />
          <textarea
            id="message"
            name="message"
            onChange={[Function]}
            type="text"
          />
        </label>
      </div>
    </div>
    <div
      className="send-pmt-type"
    >
      <p>
        What's this payment for?
      </p>
      <div
        className=""
        onClick={[Function]}
      >
        I'm sending money to family or friends
         
      </div>
      <div
        className=""
        onClick={[Function]}
      >
        I'm paying for goods or services
         
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <button
      className="send-footer-button"
      name="clear"
      onClick={[Function]}
    >
      Clear
    </button>
    <button
      className="send-footer-button"
      name="next"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
